envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -310,15 +323,17 @@ internal class EnvoySnapshotFactory(
         endpoints: List<ClusterLoadAssignment> = emptyList(),
         endpointsVersions: EndpointsVersion = EndpointsVersion.EMPTY_VERSION,
         routes: List<RouteConfiguration> = emptyList(),
-        routesVersion: RoutesVersion = RoutesVersion(properties.routes.initialVersion)
+        routesVersion: RoutesVersion = RoutesVersion(properties.routes.initialVersion),
+        listeners: List<Listener> = emptyList(),
+        listenersVersion: ListenersVersion = ListenersVersion.EMPTY_VERSION
```

Comment:
We shouldn't set always the same version - "empty", because this way changes in Listeners will not propagate to Envoys after EC deployment.

I see two options:
1. Create a property similar to `properties.routes.initialVersion`. If we change something in listeners, we would have to remember to also change that property. This is no not an ideal situation but we already do it for routes.
2. Generate random "empty version" when EC starts. This has a big drawback - all listeners will be updated on every EC deployment, even if nothing changed.

Reply:
Why would we want to change listeners during EC deployment? Right now this is just a copy of a static listener, we only set host/port and they change only during app/envoy deployment - not ECs.

Reply:
Listener is a big part of configuration. It consinsts of listener, filter chains, http_connection_manager.

We will for sure change it many times in the future. We may add new options to connection manager, we will enable RBAC, etc, etc.

Reply:
Of course, but now we do not need to change them. I can imagine in the future there is going to be a ui option to (for example) add delay filter and that would trigger change in filters but for *now* we do not need that. Maybe I'm misunderstanding something. We can talk offline.

Reply:
Of course, but now we do not need to change them. I can imagine in the future there is going to be a ui option to (for example) add delay filter and that would trigger change in filters but for *now* we do not need that. Maybe I'm misunderstanding something. We can talk offline.

Reply:
How do you imagine for example changing of idle_timeout on http_connection_manager? We will deploy EC and then wait till all Envoy will be restarted?

Maybe it's a good idea, but I think we should at least have an option to force immediate update of all Envoy. We have that option for routes (`properties.routes.initialVersion`), we don't have this option for listeners?

Reply:
@pzmi @pbetkier @lukidzi @chemicL vote for option 1 or 2, I think that we should go with 1, gives us control when it happens and incremental changes can be introduced by deploying a service.